src/ray/common/ray_syncer/ray_syncer.cc

```diff
@@ -197,23 +199,20 @@ std::shared_ptr<const RaySyncMessage> RaySyncer::GetSyncMessage(
 }
 
 std::vector<std::string> RaySyncer::GetAllConnectedNodeIDs() const {
-  std::promise<std::vector<std::string>> promise;
-  io_context_.dispatch(
-      [&]() {
-        std::vector<std::string> nodes;
-        for (auto [node_id, _] : sync_reactors_) {
-          nodes.push_back(node_id);
-        }
-        promise.set_value(std::move(nodes));
-      },
-      "");
-  return promise.get_future().get();
+  auto task = std::packaged_task<std::vector<std::string>()>([&]() {
+    std::vector<std::string> nodes;
+    for (auto [node_id, _] : sync_reactors_) {
+      nodes.push_back(node_id);
+    }
+    return nodes;
+  });
+  return boost::asio::dispatch(io_context_.get_executor(), std::move(task)).get();
```

Comment:
@rkooo567 it's for better way to write blocking call. 
I think `io_context_.dispatch` is the old style  and asio now has executor.

I don't think there is big difference. Just playing with new things so we'll be able to get better idea of how to write asio code (we highly depend on asio and I feel we are not using it in a correct way :( )

Reply:
I think future actually should be returned so the caller can decide what to do.

A better way is to accept a completion token and the caller can decide what to do with that. I'd say we can adopt this pattern in gRPC refactoring work in the future.